### PR TITLE
fixed remote query issue

### PR DESCRIPTION
**Pull request by mohitNEC**

### CODE REVIEW

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/constants/NGSIConstants.java [hunk 1]
```diff
@@ -295,4 +295,5 @@ public interface NGSIConstants {
 	public static final String NGSI_LD_SCOPE = "https://uri.etsi.org/ngsi-ld/default-context/scope";
 	public static final String NGSI_LD_SCOPE_Q = "https://uri.etsi.org/ngsi-ld/default-context/scopeQ";
 	public static final Object NGSI_LD_TENANT_SHORT = "tenant";
+	public static final String CONTENT_TYPE_HEADER = "Content-Type";
```
Comment (amitrnec) [inline Commons/src/main/java/eu/neclab/ngsildbroker/commons/constants/NGSIConstants.java]: No need to declare again. It is already in AppConstants constant class.

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/controllers/QueryControllerFunctions.java [hunk 2]
```diff
@@ -58,7 +58,7 @@ public static ResponseEntity<String> getEntity(EntryQueryService queryService, H
 			return ResponseEntity.status(HttpStatus.BAD_REQUEST).header(HttpHeaders.CONTENT_TYPE, AppConstants.NGB_APPLICATION_JSON)
 					.body(new RestResponse(ErrorType.BadRequestData, "id is not a URI").toJson());
 		}
-		String originalQuery = NGSIConstants.QUERY_PARAMETER_ID + "=" + entityId;
+		String originalQuery = entityId;
```
Comment (amitrnec) [inline Commons/src/main/java/eu/neclab/ngsildbroker/commons/controllers/QueryControllerFunctions.java:61]: Please test properly and add this test cases in FT sheet.

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/requests/UpdateEntityRequest.java [hunk 3]
```diff
@@ -9,11 +10,14 @@
 import com.github.jsonldjava.utils.JsonUtils;
 import com.google.common.collect.ArrayListMultimap;
```
Comment (amitrnec) [inline Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/requests/UpdateEntityRequest.java:12]: Remove space

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/requests/UpdateEntityRequest.java [hunk 4]
```diff
@@ -57,7 +61,12 @@ private UpdateResult updateFields(Map<String, Object> entityBody, Map<String, Ob
 		String now = SerializationTools.formatter.format(Instant.now());
 		UpdateResult updateResult = new UpdateResult();
 
+		System.out.println("Print Data --- " + attrId);
```
Comment (amitrnec) [inline Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/requests/UpdateEntityRequest.java]: Remove SOP

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/requests/UpdateEntityRequest.java [hunk 5]
```diff
@@ -57,7 +61,12 @@ private UpdateResult updateFields(Map<String, Object> entityBody, Map<String, Ob
...
+		System.out.println("Print Body --- " + entityBody.containsKey(attrId));
+		System.out.println("Print Body1 --- " + entityBody);
+		
 		if (attrId != null) {
+//			if(!isResponseNull) {
```
Comment (amitrnec) [inline Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/requests/UpdateEntityRequest.java]: Remove comment code and fix indentation

File: Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/controller/EntityController.java [hunk 6]
```diff
@@ -143,15 +146,24 @@ public ResponseEntity<String> partialUpdateEntity(HttpServletRequest request,
 				}
 			}
 			String expandedAttrib = ParamsResolver.expandAttribute(attrId, context);
-
-			UpdateResult update = entityService.partialUpdateEntity(HttpUtils.getHeaders(request), entityId,
+			
+			ResponseEntity<String> up = entityService.patchtoEndpoint(entityId, HttpUtils.getHeaders(request), payload, attrId);
+			
+			System.out.println("resp --- " + up);
```
Comment (amitrnec) [inline Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/controller/EntityController.java]: Remove SOP

File: Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/controller/EntityController.java [hunk 7]
```diff
@@ -143,15 +146,24 @@ public ResponseEntity<String> partialUpdateEntity(HttpServletRequest request,
 				}
 			}
 			String expandedAttrib = ParamsResolver.expandAttribute(attrId, context);
-
-			UpdateResult update = entityService.partialUpdateEntity(HttpUtils.getHeaders(request), entityId,
+			
+			ResponseEntity<String> up = entityService.patchtoEndpoint(entityId, HttpUtils.getHeaders(request), payload, attrId);
```
Comment (amitrnec) [inline Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/controller/EntityController.java]: Rename "up" use conventional name

File: Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/controller/EntityController.java [hunk 8]
```diff
@@ -143,15 +146,24 @@ public ResponseEntity<String> partialUpdateEntity(HttpServletRequest request,
...
+				UpdateResult update = entityService.partialUpdateEntity(HttpUtils.getHeaders(request), entityId,
 					expandedAttrib, expandedPayload);
-			logger.trace("partial-update entity :: completed");
-			if (update.getNotUpdated().isEmpty()) {
-				return ResponseEntity.noContent().build();
-			} else {
-				return HttpUtils.handleControllerExceptions(new ResponseException(ErrorType.BadRequestData, JsonUtils
-						.toPrettyString(JsonLdProcessor.compact(update.getNotUpdated().get(0), context, opts))));
+				logger.trace("partial-update entity :: completed");
+
```
Comment (amitrnec) [inline Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/controller/EntityController.java]: Remove space and fix indentation

File: Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/controller/EntityController.java [hunk 9]
```diff
@@ -162,6 +174,7 @@ public ResponseEntity<String> partialUpdateEntity(HttpServletRequest request,
 		} catch (Exception exception) {
 			return HttpUtils.handleControllerExceptions(exception);
 		}
+		return null;
```
Comment (amitrnec) [inline Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/controller/EntityController.java]: It should be return 204

File: Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/services/EntityService.java [hunk 10]
```diff
@@ -256,7 +267,11 @@ public UpdateResult partialUpdateEntity(ArrayListMultimap<String, String> header
 		// get entity details
 		Map<String, Object> entityBody = validateIdAndGetBody(entityId, tenantid);
 
+		System.out.println("Print EntityBody ---- " + entityBody);
```
Comment (amitrnec) [inline Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/services/EntityService.java]: Remove SOP

File: Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/services/EntityService.java [hunk 11]
```diff
@@ -267,6 +282,39 @@ public UpdateResult partialUpdateEntity(ArrayListMultimap<String, String> header
...
+			HttpComponentsClientHttpRequestFactory requestFactory = new HttpComponentsClientHttpRequestFactory();
+			RestTemplate restTemplate = new RestTemplate(requestFactory);
+
+			logger.debug("url " + endpoint + "/ngsi-ld/v1/entities/" + entityId + "attrs" + attrId);
+			HttpEntity<String> httpEntity = new HttpEntity<>(payload, header);
+			String patchuri = endpoint + "/ngsi-ld/v1/entities/" + entityId + "/attrs/" + attrId;
+			
+			System.out.println("Print PatchURI --- " + patchuri);
+			
+			res = restTemplate.exchange(patchuri, HttpMethod.PATCH, httpEntity,
```
Comment (amitrnec) [inline Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/services/EntityService.java]: Please give convenient name

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/constants/NGSIConstants.java (hunk 1 above)
Comment (mohitNEC) [reply Commons/src/main/java/eu/neclab/ngsildbroker/commons/constants/NGSIConstants.java]: Done

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/controllers/QueryControllerFunctions.java (hunk 2 above)
Comment (mohitNEC) [reply Commons/src/main/java/eu/neclab/ngsildbroker/commons/controllers/QueryControllerFunctions.java:61]: Done

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/requests/UpdateEntityRequest.java (hunk 3 above)
Comment (mohitNEC) [reply Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/requests/UpdateEntityRequest.java:12]: Done

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/requests/UpdateEntityRequest.java (hunk 4 above)
Comment (mohitNEC) [reply Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/requests/UpdateEntityRequest.java]: Done

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/requests/UpdateEntityRequest.java (hunk 5 above)
Comment (mohitNEC) [reply Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/requests/UpdateEntityRequest.java]: Done

File: Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/controller/EntityController.java (hunk 6 above)
Comment (mohitNEC) [reply Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/controller/EntityController.java]: Done

File: Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/controller/EntityController.java (hunk 7 above)
Comment (mohitNEC) [reply Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/controller/EntityController.java]: Done

File: Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/controller/EntityController.java (hunk 8 above)
Comment (mohitNEC) [reply Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/controller/EntityController.java]: Done

File: Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/controller/EntityController.java (hunk 9 above)
Comment (mohitNEC) [reply Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/controller/EntityController.java]: Done

File: Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/services/EntityService.java (hunk 10 above)
Comment (mohitNEC) [reply Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/services/EntityService.java]: Done

File: Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/services/EntityService.java (hunk 11 above)
Comment (mohitNEC) [reply Core/EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/services/EntityService.java]: Done

**Pull request merged**